internal/app/synchronizer/evaluator_client.go

```diff
@@ -0,0 +1,20 @@
+package synchronizer
+
+import (
+	"open-match.dev/open-match/pkg/pb"
+)
+
+// EvaluatorFunction is the signature of the function that the synchronizer
```

Comment:
Since I had evaluator_client implementation ready, rather than adding more detailed TODOS. I added the implementation and made all changes here as to the final state. Please review this file again. Note that some other changes needed for E2E test (actual config values in yaml etc.) will still follow in a separate PR.